stats/worker.go

```diff
@@ -132,37 +131,28 @@ func (v *View) Unregister() error {
 	return <-req.err
 }
 
-// Subscribe subscribes a channel to a View. If the view wasn't already
-// registered, it will be automatically registered. It allows for many clients
-// to consume the same ViewData with a single registration. -i.e. the aggregate
-// of the collected measurements will be reported to the calling code through
-// channel c. To avoid data loss, clients must ensure that channel sends
-// proceed in a timely manner. The calling code is responsible for using a
-// buffered channel or blocking on the channel waiting for the collected data.
-func (v *View) Subscribe(c chan *ViewData) error {
-	if v == nil {
-		return errors.New("cannot subscribe nil view")
-	}
+// Subscribe subscribes a view. Once a view is subscribed, it reports data
+// via the exporters.
+// During subscription, if the view wasn't registered, it will be automatically
+// registered. Once the view no longer needed to export data, user should unsubscribe
+// the view.
+func (v *View) Subscribe() error {
 	req := &subscribeToViewReq{
 		v:   v,
-		c:   c,
 		err: make(chan error),
 	}
 	defaultWorker.c <- req
 	return <-req.err
 }
 
-// Unsubscribe unsubscribes a previously subscribed channel from the
-// View subscriptions. If no more subscriber for v exists and the the ad hoc
+// Unsubscribe unsubscribes a previously subscribed channel.
```

Comment:
No more channels correct?

Reply:
The external APIs won't work with channels anymore.